README.md

```diff
@@ -257,7 +257,69 @@ Add your gateway by providing a name and connection profile via the `Add Gateway
 You can also create a gateway from a fabric environment. When you run the `Add Gateway` command there will be an option to create a gateway from a fabric environment, select this then choose the environment you want to create the gateway from.
 
 ### Connect to a gateway and discover its resources
-Connect by clicking on a gateway in the `Fabric Gateways` panel, and expand the navigation tree to explore its resources. Instantiated Smart Contracts are listed under the channel and from here you can generate functional-level test files on single or multiple smart contracts (Currently you cannot generate Java functional tests). Submit or evaluate individual transactions listed under the instantiated smart contracts, with the result displayed in the `Blockchain` output channel.
+Connect by clicking on a gateway in the `Fabric Gateways` panel, and expand the navigation tree to explore its resources. Instantiated Smart Contracts are listed under the channel and from here you can generate functional-level test files on single or multiple smart contracts. Submit or evaluate individual transactions listed under the instantiated smart contracts, with the result displayed in the `Blockchain` output channel.
+
+#### _<font color='red'>* Beta * </font> Java functional tests_
+When creating Java functional tests new dependencies must be added to the build file.
+
+If you select 'Yes' when prompted with `The last step might overwrite build.gradle/pom.xml. Do you wish to continue?`, these modifications will be done automatically. You can alternatively choose to skip this step and manually add the code listed below.
+
+##### Gradle Project - modify build.gradle:
+
+```
+Add the following repository:
```

Comment:
Could we tidy up this formatting with:
---

##### Gradle Project - modify build.gradle:

Add the following repository:
```
maven {
    url "https://oss.sonatype.org/content/repositories/snapshots"
}
```

Add the following dependencies:
```
testImplementation 'org.hyperledger.fabric:fabric-gateway-java:1.4.1-SNAPSHOT'
testImplementation 'org.assertj:assertj-core:3.14.0'
testImplementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.10.0'
testImplementation 'com.fasterxml.jackson.core:jackson-databind:2.10.0'
```

##### Maven Project - modify pom.xml:

Add the following repository:
```
<repository>
    <id>nexus</id>
    <url>https://oss.sonatype.org/content/repositories/snapshots</url>
</repository>
```
Add the following dependencies:
```
<dependency>
    <groupId>org.hyperledger.fabric</groupId>
    <artifactId>fabric-gateway-java</artifactId>
    <version>1.4.1-SNAPSHOT</version>
</dependency>
<dependency>
    <groupId>org.assertj</groupId>
    <artifactId>assertj-core</artifactId>
    <version>3.14.0</version>
</dependency>
<dependency>
    <groupId>com.fasterxml.jackson.dataformat</groupId>
    <artifactId>jackson-dataformat-yaml</artifactId>
    <version>2.10.0</version>
</dependency> 
<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-databind</artifactId>
    <version>2.10.0</version>
</dependency>
```